python/modules/IcePy/Slice.cpp

```diff
@@ -108,6 +115,12 @@ IcePy_loadSlice(PyObject* /*self*/, PyObject* args)
 
     for (const auto& fragment : compilationResult.fragments)
     {
+        // Skip if the Python module for this fragment was already loaded
+        if (loadedFragments.find(fragment.moduleName) != loadedFragments.end())
```

Comment:
A consequence of this update (+ insert on line 178) is if two or more fragments in the compilationResult have the same moduleName, we now skip the fragments number 2 and up.

If each moduleName is expected to be unique within a compilationResult, it would be nice to add a comment on moduleName to that effect.

Reply:
`VisitorCenter.Greeter`